automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -346,4 +352,30 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     val preemptibleNodePrefix = clusterName.string + "-sw"
     notebookPage.executeCell(s"! grep Finished ~/hail.log | grep $preemptibleNodePrefix").get should include(preemptibleNodePrefix)
   }
+
+  def uploadDownloadTest(cluster: Cluster, fileName: String, timeout: FiniteDuration)(assertion: (File, File) => Any)(implicit webDriver: WebDriver, token: AuthToken): Any = {
+    val uploadFile = ResourceFile(s"diff-tests/$fileName")
+    uploadFile.exists() shouldBe true
+
+    withNotebookUpload(cluster, uploadFile) { notebook =>
+      notebook.runAllCells(timeout.toSeconds)
+      notebook.download()
+    }
+
+    // sanity check the file downloaded correctly
+    val downloadFile = new File(downloadDir, "utf-8''"+fileName)  // TODO https://github.com/DataBiosphere/leonardo/issues/214
```

Comment:
Data point: the issue in https://github.com/DataBiosphere/leonardo/issues/214 happens in auto tests as well